spec/unit/files/servicenow_spec.rb

```diff
@@ -188,10 +199,40 @@
         end
 
         it 'decrypts the password' do
-          expect(ServiceNowRequest).to receive(:new).with(anything, anything, anything, 'admin', 'password')
+          expect(ServiceNowRequest).to receive(:new).with(anything, anything, anything, 'admin', 'password', 'oauth_token')
           expect { servicenow('example.puppet.com') }.to raise_error(NoMethodError)
         end
       end
     end
+
+    context 'with hiera-eyaml encrypted oauth_token' do
+      # Note: This oauth_token is the encrypted form of 'oauth_token'. It was generated by the command
+      # 'eyaml encrypt -s 'oauth_token' --pkcs7-private-key=./spec/support/files/private_key.pkcs7.pem --pkcs7-public-key=./spec/support/files/public_key.pkcs7.pem'
+      let(:encrypted_oauth_token) do
+        'ENC[PKCS7,MIIBeQYJKoZIhvcNAQcDoIIBajCCAWYCAQAxggEhMIIBHQIBADAFMAACAQEwDQYJKoZIhvcNAQEBBQAEggEAKVAUJvBJGrG25SGq0oVymzCxlQ3rvnqNHvl4rKagNshNDe0FKXUxDv0lz/DuklYMTFnKrm8gZxNESvr35ecBM2FckDy1NkIaWWKVFMg5H7KuZaCN/mFgtEpwUkUl3yJpcoJsfN4FpdCWAwjLF1qdOQ25nMEB9sKezZUKMjKm0pnGslr2Gj35HTTxc78HgT9cgVZHi5+NefFlMHDUZWyuSeL4xr4msUFDn6F1RoJp8zYPz31kBMgbowTNxICJV4vX8plwNgLcJicuqeOsEkznO/1bc+fh2yyiAUqimwctd20oni6eubkV8JY5wxfETX+GOiHuHCYZPFemTXHxl3O/GTA8BgkqhkiG9w0BBwEwHQYJYIZIAWUDBAEqBBBl5Z3XF8s8RfEEGTDABqwDgBAadR7I9hBGLSC0m5Ut6xzo]'
+      end
+      let(:config) do
+        default_config = super()
+        default_config['oauth_token'] = encrypted_oauth_token
+        default_config
+      end
+
+      before(:each) do
```

Comment:
Nit: you _could_ refactor this setup into something like
```
shared_context 'hiera-eyaml setup' do
  before(:each) do
    ...
  end
end
```

Then the password and oauth token eyaml tests can be structured as

```
context 'encrypted thing' do
  include_context 'hiera-eyaml setup'

  ...
end
```

Reply:
But this comment is not a blocker so if you don't want to do it/think it isn't needed, then I'm :thumbsup: on that too.